src/number/n_algExt.jl

```diff
@@ -0,0 +1,329 @@
+export n_algExt, N_AlgExtField, AlgebraicExtensionField, modulus
+
+###############################################################################
+#
+#   Data type and parent methods
+#
+###############################################################################
+
+elem_type(::Type{N_AlgExtField}) = n_algExt
+
+parent(a::n_algExt) = a.parent
+
+parent_type(::Type{n_algExt}) = N_AlgExtField
+
+base_ring(a::n_algExt) = base_ring(parent(a))
+
+base_ring(a::N_AlgExtField) = base_ring(a.minpoly)
+
+@doc Markdown.doc"""
+    gen(F::N_AlgExtField)
+
+Return the generator the given algebraic extension.
+"""
+function gen(F::N_AlgExtField)
+   return F(libSingular.n_Param(Cint(1), F.ptr))
+end
+
+@doc Markdown.doc"""
+    characteristic(R::N_AlgExtField)
+
+Return the characteristic of the field.
+"""
+function characteristic(R::N_AlgExtField)
+   return ZZ(libSingular.n_GetChar(R.ptr))
+end
+
+function deepcopy_internal(a::n_algExt, dict::IdDict)
+   return parent(a)(libSingular.n_Copy(a.ptr, parent(a).ptr))
+end
+
+function hash(a::n_algExt, h::UInt)
+   error("hash not implemented")
```

Comment:
You sure about this one?

Reply:
proper implementation needs more help from the binaries. maybe next time around.

Reply:
Making it not crash is a good start.